src/Microsoft.AspNet.OData.Shared/TypelessPatchMethodHandler.cs

```diff
@@ -0,0 +1,196 @@
+        {
+            this.entityType = entityType;
+            this.originalList = originalList?? new List<EdmStructuredObject>();
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override PatchStatus TryGet(IDictionary<string, object> keyValues, out EdmStructuredObject originalObject, out string errorMessage)
+        {
+            PatchStatus status = PatchStatus.Success;
+            errorMessage = string.Empty;
+            originalObject = default(EdmStructuredObject);
+
+            try
+            {
+                originalObject = GetFilteredItem(keyValues);
+
+                if (originalObject == null)
+                {
+                    status = PatchStatus.NotFound;
+                }
+            }
+            catch (Exception ex)
+            {
+                status = PatchStatus.Failure;
+                errorMessage = ex.Message;
+            }
+
+            return status;
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override PatchStatus TryCreate(IEdmChangedObject changedObject, out EdmStructuredObject createdObject, out string errorMessage)
+        {
+            createdObject = default(EdmStructuredObject);
+            errorMessage = string.Empty;
+
+            try
+            {
+                if(originalList == null)
+                {
+                    originalList = new List<EdmStructuredObject>();
+                }
+
+                createdObject = new EdmEntityObject(entityType);
+                originalList.Add(createdObject);
+
+                return PatchStatus.Success;
+            }
+            catch (Exception ex)
+            {
+                errorMessage = ex.Message;
+
+                return PatchStatus.Failure;
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override PatchStatus TryDelete(IDictionary<string, object> keyValues, out string errorMessage)
+        {
+            errorMessage = string.Empty;
+
+            try
+            {
+                EdmStructuredObject originalObject = GetFilteredItem(keyValues);
+                originalList.Remove(originalObject);
+
+                return PatchStatus.Success;
+            }
+            catch (Exception ex)
+            {
+                errorMessage = ex.Message;
+
+                return PatchStatus.Failure;
+            }
+        }
+
+        public override TypelessPatchMethodHandler GetNestedPatchHandler(EdmStructuredObject parent, string navigationPropertyName)
+        {
+            IEdmNavigationProperty navProperty = entityType.DeclaredNavigationProperties().FirstOrDefault(x => x.Name == navigationPropertyName);
```

Comment:
>DeclaredNavigationProperties [](start = 60, length = 28)

DeclaredNavigationProperties() only returns navigation properties defined in this derived type.  Use NavigationProperties() to include navigation properties defined in any base types.